front/lib/api/assistant/actions/retrieval.ts

```diff
@@ -546,7 +607,17 @@ export class RetrievalConfigurationServerRunner extends BaseActionConfigurationS
             token_count: number;
           }[];
 
-          const refs = getRefs().slice(refsOffset, refsOffset + v.length);
+          if (refsOffset + topK > getRefs().length) {
+            // This is a stream dropping error since the guardrails put in place should prevent this
+            // from ever happeaning (max 16 actions per step and sharing of topK among actions of
+            // the same type).
+            throw new Error(
```

Comment:
let's also panic ? if this starts happening we need to know ASAP

Reply:
This triggers a stream dropping error which is ~~~ the highest level of panic we have today